File: app/src/main/java/com/mxt/anitrend/base/interfaces/base/BaseAnimation.java
package com.mxt.anitrend.base.interfaces.base;

import android.animation.Animator;
import android.view.View;
import android.view.animation.Interpolator;

/**
 * Created by max on 2018/02/24.
 * original impl -> https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */

public interface BaseAnimation {

    int DEFAULT_ANIMATION_DURATION = 250;

    Animator[] getAnimators(View view);

    Interpolator getInterpolator();

    int getAnimationDuration();
}

File: app/src/main/java/com/mxt/anitrend/base/interfaces/base/BaseAnimation.kt
package com.mxt.anitrend.base.interfaces.base

import android.animation.Animator
import android.view.View
import android.view.animation.Interpolator

/**
 * Created by max on 2018/02/24.
 * original impl -> https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */
interface BaseAnimation {
    companion object {
        const val DEFAULT_ANIMATION_DURATION = 250
    }

    fun getAnimators(view: View): Array<Animator>

    fun getInterpolator(): Interpolator

    fun getAnimationDuration(): Int
}
